### PR TITLE
Code for get-api-article_id-comments

### PR DESCRIPTION
**Pull request by mejaz4**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -5,98 +5,139 @@ const seed = require('../db/seeds/seed.js')
...
+describe('4. GET /api/articles/:article_id/comments', () => {
+  test('status:200, responds with comments from a single matching article_id', () => {
+    const articleId = 3;
+    return request(app)
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body
+        expect(comments).toBeInstanceOf(Array);
+        comments.forEach((comment) => {
```
Comment (lewispricey) [inline __tests__/app.test.js:114]: It would be good to add an assertion to check the array length is what we'd expect to prevent false positives if we'd expect comments but the array is empty

File: __tests__/app.test.js [hunk 2]
```diff
@@ -5,98 +5,139 @@ const seed = require('../db/seeds/seed.js')
...
+});
 
-    test('404: valid article_id but doesnt exist', () => {
-        return request(app)
-        .get('/api/articles/30')
-        .expect(404)
-        .then((response) => {
-            const msg = response.body.msg;
-            expect(msg).toBe('Not Found')
+describe('4. GET /api/articles/:article_id/comments', () => {
```
Comment (lewispricey) [inline __tests__/app.test.js:104]: Please could you add a test for an empty array if the article exists but doesn't have comments

File: __tests__/app.test.js [hunk 3]
```diff
@@ -5,98 +5,139 @@ const seed = require('../db/seeds/seed.js')
...
+  test('404: valid article_id but doesnt exist', () => {
+    return request(app)
+      .get('/api/articles/30/comments')
+      .expect(404)
+      .then((response) => {
+        const msg = response.body.msg;
+        expect(msg).toBe('Not Found')
+      })
+  })
+  test('400: invalid article_id', () => {
```
Comment (lewispricey) [inline __tests__/app.test.js:146]: This is a really good test

File: controllers/controller.articles.js [hunk 4]
```diff
@@ -17,4 +17,14 @@ const getArticleById  = (req, res, next) => {
     });
   };
 
-module.exports = {getArticles, getArticleById}
+const getArticleComments = (req, res, next) => {
+  const article_id = req.params.article_id;
+  selectArticleComments(article_id).then((comments) => {
+    res.status(200).send({ comments });
```
Comment (lewispricey) [inline controllers/controller.articles.js]: Good choice of key to send the data back on 👍

File: models/model.articles.js [hunk 5]
```diff
@@ -25,4 +25,17 @@ const selectArticleById = (article_id) => {
...
 
-module.exports = { selectArticles, selectArticleById }
\ No newline at end of file
+const selectArticleComments = (article_id) => {
+    const SQL = `SELECT comment_id, comments.votes, comments.created_at, comments.author, comments.article_id, comments.body FROM comments
+    WHERE article_id = $1
+    ORDER BY comments.created_at DESC;`;
+
+    return db.query(SQL, [article_id]).then((results) => {
+        if (results.rowCount === 0) {
```
Comment (lewispricey) [inline models/model.articles.js]: Would this cause a false error if the article exists but has no comments?

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work so far Musa, Just an extra test and assertion to add in.

Please raise another PR once the changes have been made.

**Review by leekli — APPROVED** [review]
Thanks for making these changes, Musa. Well done.

You are free to merge this branch.
